projects/target-platforms/packages/create-plaform/bin/create-platform.js

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env node
```

Comment:
So this file essentially generates a `package.json` and `config.json` to a new directory? And this would be the package we would publish to npm?

btw I am reading this through commit by commit, so things might change in future commits I'm not aware of yet.

Reply:
Yes, it generates a new platform project next to the place where the script lives. So, the idea is that we run `yarn link` inside the `create-platform` project and then invoke it like:

`create-platform /path/to/your/liferay-portal portal-N.N-gaN`

I didn't put too much safeguards into the script because we will be the only ones running it so I think explaining how it works in the README or CONTRIBUTING files should be enough. Also, it is very probable that we may need to tweak it every time it is run because of little changes to the `liferay-portal` source tree, so I expect it to evolve every time we run it so I consider it more a helping tool than a real stable tool that anyone could use (at least for now).

I was not going to create a script at first, but then I tried to define the platform by hand and it was so complex and error prone that I decided to automate it (especially to avoid human errors).

Additionally, the `create-platform` script contains [assets](https://github.com/liferay/liferay-frontend-projects/pull/546/files#diff-8cbe038698a422a3fbb09f05c95e2c11d84cf9b9cdf7c1903d66b1f876475d3d) that are copied to every plaform it generates. I don't know if this structure will last long, but it suffices for now.

The main goal is to have a centralized point where we can manage these things, lint them, etc, but without needing to maintain another npm package to be shared by all platforms. I think this structure is better because having a shared package for all platforms looks like something quite difficult (given how much we change the build process from version to version) but if we stabilize portal's build enough in the future we can switch to that model.

In any case, the good thing is that everything is kept under the "platform" cover, so we may change anything we like in the future without impacting users.

Right now, the only APIs we are publishing are:

- The name of the platform dependency
- The name of the build script and its API (`build` command for now)

I think those are the only two things we won't be able to change in the future without incurring in a breaking change.

Reply:
I've added [this commit](https://github.com/liferay/liferay-frontend-projects/pull/546/commits/25b88ebcdb9f209787da9f49d5a40862af1ddaca) to explain it all.